### PR TITLE
Webhookevent client

### PR DESCRIPTION
**Pull request by shalper2**
**Description:** Implemented the component. It works by standing up a server with a route to accept post requests from the webhook its configured to receive from. These requests are then converted into otel logs and shipped. Can handle gzip encoded payloads.

**Link to tracking Issue:** [18101](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18101)

**Testing:** Basic unit testing for each code section

**Documentation:** Documentation is largely unchanged. Proper readme will likely be included in a separate PR

### CODE REVIEW

**Comment by shalper2** [comment]
Curious if anyone has thoughts about what to add as attributes to the log

File: receiver/webhookeventreceiver/receiver.go
Comment (atoulme) [inline receiver/webhookeventreceiver/receiver.go:42]:
```suggestion
```

File: receiver/webhookeventreceiver/receiver.go
Comment (atoulme) [inline receiver/webhookeventreceiver/receiver.go:200]: make sure to also call EndLogsOp in the failure case

File: receiver/webhookeventreceiver/req_to_log.go
Comment (atoulme) [inline receiver/webhookeventreceiver/req_to_log.go:31]: If you want, and that can be a separate PR, you can add metadata to the Scope of the scopeLog.

File: receiver/webhookeventreceiver/req_to_log.go
Comment (atoulme) [inline receiver/webhookeventreceiver/req_to_log.go:51]: this is better added in the scope log scope.

**Review by atoulme — COMMENTED** [review]
Nice code, some nits, please see my comments.

File: receiver/webhookeventreceiver/req_to_log.go
Comment (shalper2) [reply receiver/webhookeventreceiver/req_to_log.go:31]: For sure! I think my approach to this is to iterate as it becomes more clear what type of information users of the tool expect to be shipped with their logs. I didn't want to add a bunch of stuff that wasn't important to anyone off the bat.

File: receiver/webhookeventreceiver/req_to_log.go
Comment (atoulme) [inline receiver/webhookeventreceiver/req_to_log.go:36]: we can revisit this in a bit

**Review by atoulme — APPROVED** [review]
Approved. There is a nit with Scope, you can do:
scope.SetName(...)
scope.SetVersion(...)

This info is available to you in the CreateSettings object passed in.

Example: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/dde97b55b671af44c6f02a4ad0fecf6c9b65f68e/receiver/azureeventhubreceiver/azureresourcelogs_unmarshaler.go#L109

**Comment by shalper2** [comment]
@atoulme Could we remove the azuremonitor tag from this PR? I think thats why it wont merge without further review...

File: receiver/webhookeventreceiver/config_test.go
Comment (codeboten) [inline receiver/webhookeventreceiver/config_test.go:96]: nit: please use localhost in tests to avoid listening on 0.0.0.0

File: receiver/webhookeventreceiver/receiver.go
Comment (codeboten) [inline receiver/webhookeventreceiver/receiver.go:83]:
```suggestion
// Start function manages receiver startup tasks. part of the receiver.Logs interface.
```

File: receiver/webhookeventreceiver/receiver.go
Comment (codeboten) [inline receiver/webhookeventreceiver/receiver.go:134]:
```suggestion
// Shutdown function manages receiver shutdown tasks. part of the receiver.Logs interface.
```

File: receiver/webhookeventreceiver/receiver.go
Comment (codeboten) [inline receiver/webhookeventreceiver/receiver.go:141]:
```suggestion
// handleReq handles incoming request from webhook. On success returns a 200 response code to the webhook
```

File: receiver/webhookeventreceiver/receiver_test.go
Comment (codeboten) [inline receiver/webhookeventreceiver/receiver_test.go:77]: same comment about using localhost

File: receiver/webhookeventreceiver/factory.go
Comment (shalper2) [reply receiver/webhookeventreceiver/factory.go:55]: I don't have to handle that error?

Comment (codeboten) [reply receiver/webhookeventreceiver/factory.go:55]: i mean you're handling it by returning it anyways, there's nothing happening between lines 55-60 that wouldn't happen if you just returned here

Comment (codeboten) [inline receiver/webhookeventreceiver/factory.go:55]:
```suggestion
	return newLogsReceiver(params, *conf, consumer)
```

**Comment by atoulme** [comment]
@codeboten ready for another look

**Review by codeboten — APPROVED** [review]

**Comment by shalper2** [comment]
Sorry got a little fast and loose when i was rebasing there. Might need another review before merge

**Comment by codeboten** [comment]
Collector module versions need an update in the go mod and some linting issues to sort out:

```
Error: req_to_log.go:18:2: unused-parameter: parameter '_config' seems to be unused, consider removing or renaming it as _ (revive)
	_config *Config,
	^
Error: receiver.go:84:32: unused-parameter: parameter 'ctx' seems to be unused, consider removing or renaming it as _ (revive)
func (er *eventReceiver) Start(ctx context.Context, host component.Host) error {
                               ^
Error: receiver.go:135:35: unused-parameter: parameter 'ctx' seems to be unused, consider removing or renaming it as _ (revive)
func (er *eventReceiver) Shutdown(ctx context.Context) error {
                                  ^
Error: receiver.go:206:37: unused-parameter: parameter '_ctx' seems to be unused, consider removing or renaming it as _ (revive)
func (er *eventReceiver) failBadReq(_ctx context.Context,
                                    ^
Error: receiver.go:210:2: redefines-builtin-id: redefinition of the built-in type error (revive)
	jsonResp, error := jsoniter.Marshal(err.Error())
	^
Error: receiver.go:219:3: redefines-builtin-id: redefinition of the built-in type error (revive)
		_, error := w.Write(jsonResp)
		^
```

**Comment by atoulme** [comment]
I have filed https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/23049 and will rerun the build component that failed.

**Review by codeboten — APPROVED** [review]
Thanks for your patience @shalper2!

**Pull request merged**
